admin/projects.go

```diff
@@ -207,6 +205,80 @@ func (s *Service) UpdateProject(ctx context.Context, proj *database.Project, opt
 	return proj, nil
 }
 
+// UpdateProjectVariables updates a project's variables and runs reconcile on the deployments.
+func (s *Service) UpdateProjectVariables(ctx context.Context, project *database.Project, environment string, vars map[string]string, unsetVars []string, userID string) error {
+	if len(vars) == 0 && len(unsetVars) == 0 {
+		return nil
+	}
+	txCtx, tx, err := s.DB.NewTx(ctx)
+	if err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+	defer func() {
+		_ = tx.Rollback()
+	}()
+
+	// Upsert variables
+	if len(vars) > 0 {
+		_, err = s.DB.UpsertProjectVariable(txCtx, project.ID, environment, vars, userID)
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
```

Comment:
It should return the error directly. The `status` package is gRPC-specific so should only be used in the `admin/server` package or related server-specific packages.

Even in the `admin/server` package, the `err` should be returned directly because it may also be a context cancelled error, which are not considered internal errors. Our gRPC server has error handling middleware that checks un-annotated errors against common errors (such as context cancellation), falling back to an internal status code for unknown errors.

Note that there are multiple occurrences of this issue in this file.

Reply:
Sure. Modified the errors at both places.